---
title: "R Notebook"
output: 
---

```{r}
library(data.table)

# download GTF file for ensembl 108 if not already in the required folder
if (!file.exists("gtf/gencode.v44.annotation.gff3.gz")) {
  download.file("https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_44/gencode.v44.annotation.gff3.gz", "gtf/gencode.v44.annotation.gff3.gz")
}
```

# needs to match file here https://github.com/Illumina/SpliceAI/blob/master/spliceai/annotations/grch38.txt

```{r}
# read in gtf file
gtf_file = list.files(path = 'gtf', full.names = TRUE)
gtf <- rtracklayer::import(gtf_file)
gtf_dt = as.data.table(gtf)

annotation_file = gtf_dt[type == 'transcript', .(`#NAME` = transcript_id, CHROM = seqnames, STRAND = strand, TX_START = start -1, TX_END = end)]

setorder(gtf_dt, transcript_id, start)


exon_info = gtf_dt[type == 'exon', .(`#NAME` = transcript_id, CHROM = seqnames, STRAND = strand, EXON_START = start -1, EXON_END = end)]

exon_info = exon_info[, .(EXON_START = paste0(paste(EXON_START, collapse = ','), ','),
              EXON_END = paste0(paste(EXON_END, collapse = ','), ',')), by = .(`#NAME`, CHROM, STRAND)]

annotation_file = annotation_file[exon_info, on = .(`#NAME`, CHROM, STRAND)]
fwrite(annotation_file, 'output/gencodev44.txt', quote = FALSE, sep = '\t')
```
